src/systems/hydrodynamics/Hydrodynamics.cc

```diff
@@ -340,13 +344,16 @@ void Hydrodynamics::PreUpdate(
   state(4) = localRotationalVelocity.Y();
   state(5) = localRotationalVelocity.Z();
 
-  auto dt = static_cast<double>(_info.dt.count())/1e9;
-  stateDot = (state - this->dataPtr->prevState)/dt;
+  auto dt = (double)_info.dt.count()/1e9;
+  auto alpha = 0.9;
+  stateDot = alpha * (state - this->dataPtr->prevState)/dt
+    + (1-alpha) * this->dataPtr->prevStateDot;
```

Comment:
I'm curious, what's the reason to filter `stateDot` here?

Reply:
This is to smoothen the rate of change of velocity, we can directly capture `WorldLinearAccelerationt` too.

Reply:
I'm saying because having the rolling average filter here means that `stateDot` depends on all previously visited states (rather than just the last one). I'm not sure if or how much we want to worry about this, though.

Reply:
That's a valid point. It was discussed in the Buoy Sim project that this might introduce lag and instability.
@arjo129 How should I proceed here? I'm thinking to get this in to be at par with the current changes (though I want to make sure this is generally sound and not specific for LRAUV). And have a separate effort to make this suitable for buoy as well.

Reply:
We can go ahead and remove this. It will not effect LRAUV much. Alternatively we can add an alpha. Finally I thought we were deprecating the added mass within this plugin and switching to the new added mass formulation.

Reply:
So I compared the previous implementation, the filter and `WorldLinearAcceleration` and the previous implementation is definitely the most stable. Going with that and we can iterate on it in future, thanks!